File: 875. Koko Eating Bananas.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {

        int speed = 1;
        int hours;
        while (true){
            hours =0;

            for(int pile : piles){
                hours += Math.ceil((double) pile / speed);
            }

            if(hours<=h){
                return  speed;
            }else{
                speed++;
            }
        }
    }
}

File: 875. Koko Eating Bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:

        def canKokoEat(k) -> bool:
            hLeft = h
            for pile in piles: hLeft -= ceil(pile/k)
            return hLeft >= 0

        result = float('inf')

        def bs(l,r):
            nonlocal result

            if l <= r:
                k = l + (r - l) // 2
                if canKokoEat(k): 
                    result = min(result,k)
                    bs(l,k-1)
                else:  bs(k+1,r)

        bs(1,max(piles))
        return result
        


                